sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/LogsHelperTests.cs

```diff
@@ -415,6 +415,47 @@ public void DuplicateKeysInLogRecordAttributesAndLogScope()
             Assert.Equal(expectedAttributeValue, actualAttributeValue);
         }
 
+        [Fact]
+        public void DuplicateKeysInLogRecordAttributesAndLogScope2()
+        {
+            // Arrange.
+            var logRecords = new List<LogRecord>(1);
+            using var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddOpenTelemetry(options =>
+                {
+                    options.IncludeScopes = true;
+                    options.AddInMemoryExporter(logRecords);
+                });
+            });
+
+            var logger = loggerFactory.CreateLogger("Some category");
+
+            const string expectedScopeKey = "Some scope key";
+            const string expectedScopeValue = "Some scope value";
+            const string duplicateScopeValue = "Some duplicate scope value";
+            const string duplicateScopeValue2 = "Another duplicate scope value";
+
+            // Act.
+            using (logger.BeginScope(new List<KeyValuePair<string, object>>
+            {
+                new KeyValuePair<string, object>(expectedScopeKey, expectedScopeValue),
+                new KeyValuePair<string, object>(expectedScopeKey, duplicateScopeValue),
+            }))
+            {
+                logger.LogInformation($"Some log information message. {{{expectedScopeKey}}}.", duplicateScopeValue2);
```

Comment:
Why using string interpolation?

Reply:
This approach was taken just for brevity. It keeps the end-to-end scenario within the method block. All of the tests in this class/project are using the same approach. 

These were initially written before the best practices were updated to recommend structured logging. We've never made it a priority to refactor these tests, and new tests are following the existing pattern.

I'll bring this up in our next standup to discuss if we want to spend the time to refactor this project. I think it would add value because some devs will refer to the examples in the unit tests. But we need to weigh that against covering our other test gaps.

Reply:
We should consider using the AI assistant to upgrade these code.